PhysicsTools/NanoAOD/python/taus_cff.py

```diff
@@ -30,23 +34,22 @@
     )
 
 ##################### Tables for final output and docs ##########################
-def _tauIdWPMask(pattern, choices, doc=""):
-    return Var(" + ".join(["%d * tauID('%s')" % (pow(2,i), pattern % c) for (i,c) in enumerate(choices)]), "uint8", 
-               doc=doc+": bitmask "+", ".join(["%d = %s" % (pow(2,i),c) for (i,c) in enumerate(choices)]))
-def _tauId2WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("Loose","Tight"),doc=doc)
-def _tauId3WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("Loose","Medium","Tight"),doc=doc)
-def _tauId4WPMask(pattern,doc):
-    return _tauIdWPMask(pattern, choices=("VLoose", "Loose", "Medium", "Tight"), doc=doc)
-def _tauId5WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("VLoose","Loose","Medium","Tight","VTight"),doc=doc)
-def _tauId6WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("VLoose","Loose","Medium","Tight","VTight","VVTight"),doc=doc)
-def _tauId7WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("VVLoose","VLoose","Loose","Medium","Tight","VTight","VVTight"),doc=doc)
-def _tauId8WPMask(pattern,doc):
-    return _tauIdWPMask(pattern,choices=("VVVLoose","VVLoose","VLoose","Loose","Medium","Tight","VTight","VVTight"),doc=doc)
+def _tauIdWPMask(pattern, choices, doc="", from_raw=False, wp_thrs=None):
+    if from_raw:
+        assert wp_thrs is not None, "wp_thrs argument in _tauIdWPMask() is None, expect it to be dict-like"
+        
+        var_definition = []
+        for wp_name in choices:
+            if not isinstance(wp_thrs[wp_name], float):
+                raise TypeError("Threshold for WP=%s is not a float number." % wp_name)
+            wp_definition = "(tauID('%s')>%.4f)" % (pattern, wp_thrs[wp_name])
```

Comment:
I think `%.4f` should be replaced with `%f` - e.g. for deepTau v2p1 the WP were not rounded (not intentionally, but we need to keep backward compatibility)

Reply:
but actually `from_raw` option (and therefore `%.4f`) is only called for v2p5, for v2p1 WPs are taken from MINIAOD directly. So backward compatibility shouldn't be affected by the formatting rounding, but I can change this anyway, if needed

Reply:
I tend to agree. As you can see the rounding showed up in test I preformed :)

Reply:
sure, I will change now all `.4f` to `format()`. I was a bit cautious to change `%` interface (which is quite old..) because I thought that it (maybe) could break python version backward compatibility or smth like this

Reply:
yes, I agree, this would make the interface transparent, so that the thresholds propagate from `tauIdWPsDefs.py` to the `Var()` definition without rounding surprises